packages/color-area/src/ColorArea.ts

```diff
@@ -113,48 +103,48 @@ export class ColorArea extends SpectrumElement {
 
     @property({ type: Number })
     public get x(): number {
-        return this._x;
+        return this.colorController.color.hsv.s / 100;
     }
 
     public set x(x: number) {
         if (x === this.x) {
             return;
         }
         const oldValue = this.x;
-        this._x = x;
         if (this.inputX) {
             // Use the native `input[type='range']` control to validate this value after `firstUpdate`
             this.inputX.value = x.toString();
-            this._x = this.inputX.valueAsNumber;
+            this.colorController.color.set(
+                's',
+                this.inputX.valueAsNumber * 100
+            );
+        } else {
+            this.colorController.color.set('s', x * 100);
         }
         this.requestUpdate('x', oldValue);
-        this.colorController.applyColorFromState();
     }
 
-    private _x = 1;
-
     @property({ type: Number })
     public get y(): number {
-        return this._y;
+        return this.colorController.color.hsv.v / 100;
     }
 
     public set y(y: number) {
         if (y === this.y) {
             return;
         }
         const oldValue = this.y;
-        this._y = y;
         if (this.inputY) {
             // Use the native `input[type='range']` control to validate this value after `firstUpdate`
             this.inputY.value = y.toString();
-            this._y = this.inputY.valueAsNumber;
+            this.colorController.color.set(
+                'v',
+                this.inputY.valueAsNumber * 100
+            );
         }
         this.requestUpdate('y', oldValue);
-        this.colorController.applyColorFromState();
```

Comment:
We do not need to do this because the component is getting updated by `this.requestUpdate('y', oldValue);` and the colorController is being updated by `this.colorController.color.set('v',this.inputY.valueAsNumber * 100 );`
Earlier we needed `this.colorController.applyColorFromState();` because the controller was not getting properly updated and we manually had to create a function `applyColorFromState()` and manually call it again and again when the x,y values were altered.